### PR TITLE
feat(@deepnote/blocks): Add functions to serialize .deepnote files

### PR DESCRIPTION
**Pull request by OlegWock**
Contributes to BLU-5635

## Summary by CodeRabbit

* **New Features**
  * Added standardized serialization for Deepnote files and snapshots, producing stable, consistent YAML output.

* **Chores**
  * Aligned serialization across CLI, conversion tools, and platform utilities to use the new serializers.
  * Updated internal module organization and import surfaces to support the change.

* **Tests**
  * Added comprehensive tests covering serialization/deserialization, field ordering, snapshots, and edge-case content handling.

### CODE REVIEW

File: packages/blocks/src/deepnote-file/serialize-deepnote-file.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  return stringify(normalized, yamlOptions)
+}
+
+/**
+ * Serialize a DeepnoteSnapshot to a YAML string.
+ */
+export function serializeDeepnoteSnapshot(snapshot: DeepnoteSnapshot): string {
+  // We pass object through zod schema to ensure stable fields order
+  const normalized = deepnoteSnapshotSchema.parse(snapshot)
+  return stringify(normalized, yamlOptions)
```
[the remark this answers is not in the document]
Comment (OlegWock) [reply packages/blocks/src/deepnote-file/serialize-deepnote-file.ts:29]: Not relevant

File: packages/blocks/src/deepnote-file/serialize-deepnote-file.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  defaultStringType: 'PLAIN',
+  defaultKeyType: 'PLAIN',
+} as const
+
+/**
+ * Serialize a DeepnoteFile to a YAML string.
+ */
+export function serializeDeepnoteFile(file: DeepnoteFile): string {
+  // We pass object through zod schema to ensure stable fields order
+  const normalized = deepnoteFileSchema.parse(file)
```
Comment (OlegWock) [inline packages/blocks/src/deepnote-file/serialize-deepnote-file.ts:19]: By default keys written in .yaml file in order they are added to the source object, so it's possible that manually constructed object (i.e. not parsed through zod) will have wildly different order of properties. While still valid YAML, it can produce unnecessarily big git diffs even for small changes in certain situations. So I opted in to force order of properties. 

There are two (easy) ways to do this: sort them alphabetically (`yaml` supports passing key sorting function as parameter) or parse them with zod, in which case keys will be ordered same way as in zod schema. I opted to use zod approach because order in schema is more human-friendly and in general makes more sense than alphabetic. This adds some overhead of course, but it's usually very short amount of time, and only around 1/5 of time actual serialization to yaml takes (e.g. 9ms vs 46ms for file with 20 notebooks, 100 blocks each)
